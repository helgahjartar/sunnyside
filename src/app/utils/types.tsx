export interface Weather {
   id: string;
   icon: string;
   day: string;
   main: string;
   temp: string;
   city: string;
   country: string;
   humidity: string;
}

export const ForecastType = {
   OneDay: 'weather',
   FiveDay: 'forecast',
};
