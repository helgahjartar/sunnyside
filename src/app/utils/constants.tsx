export const API_KEY = process.env.NEXT_PUBLIC_API_KEY;
export const BASE_URL = 'https://api.openweathermap.org/data/2.5/';
export const ICON_BASE_URL = 'https://openweathermap.org/img/wn/';
export const DEFAULT_CITY = 'Reykjav√≠k';
export const DEFAULT_COUNTRY = 'IS';
