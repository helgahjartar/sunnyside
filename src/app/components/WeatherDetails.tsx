import Image from 'next/image';
import Humidity from '../../../public/static/icons/Humidity';
import Temperature from '../../../public/static/icons/Temperature';
import Icon from './WeatherIcon';
import { Weather } from '../utils/types';
import { ICON_BASE_URL } from '../utils/constants';

const WeatherDetails = ({ weather }: { weather: Weather }) => {
   return (
      <div data-testid="weather-details">
         <div className="text-xl font-bold text-black underline">{weather.day}</div>
         <div className="flex flex-row space-x-1 items-center">
            <Icon>
               <Image
                  width={40}
                  height={40}
                  alt="Weather Icon"
                  src={`${ICON_BASE_URL}${weather.icon}@2x.png`}
               />
            </Icon>
            <div className="text-lg text-black">{weather?.main}</div>
         </div>
         <div className="flex flex-row space-x-1 items-center">
            <Icon>
               <Temperature />
            </Icon>
            <div className="text-lg text-black">{weather?.temp}Â°</div>
         </div>
         <div data-testid="humidity" className="flex flex-row space-x-1 items-center">
            <Icon>
               <Humidity />
            </Icon>
            <div className="text-lg text-black">{weather?.humidity} %</div>
         </div>
      </div>
   );
};

export default WeatherDetails;
