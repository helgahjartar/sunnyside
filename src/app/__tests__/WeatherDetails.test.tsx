import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import WeatherDetails from '../components/WeatherDetails';
import { Weather } from '../utils/types';

describe('WeatherDetails', () => {
   test('matches snapshot based on weather passed down', () => {
      const mockWeather: Weather = {
         id: '1',
         icon: 'mockUrl',
         day: 'Sunday',
         main: 'Tokyo',
         temp: '25',
         humidity: '97',
      };

      render(<WeatherDetails weather={mockWeather} />);

      expect(screen.getByTestId('weather-details')).toMatchSnapshot();
   });
});
