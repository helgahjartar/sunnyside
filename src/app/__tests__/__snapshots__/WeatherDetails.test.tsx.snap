// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDetails matches snapshot based on weather passed down 1`] = `
<div
  data-testid="weather-details"
>
  <div
    class="text-xl font-bold text-black underline"
  >
    Sunday
  </div>
  <div
    class="flex flex-row space-x-1 items-center"
  >
    <div
      class="h-10 w-10 undefined"
    >
      <img
        alt="Weather Icon"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fopenweathermap.org%2Fimg%2Fwn%2FmockUrl%402x.png&w=96&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fopenweathermap.org%2Fimg%2Fwn%2FmockUrl%402x.png&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Fopenweathermap.org%2Fimg%2Fwn%2FmockUrl%402x.png&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
    </div>
    <div
      class="text-lg text-black"
    >
      Tokyo
    </div>
  </div>
  <div
    class="flex flex-row space-x-1 items-center"
  >
    <div
      class="h-10 w-10 undefined"
    >
      <svg
        viewBox="0 0 30 30"
        xml:space="preserve"
      >
        <path
          d="M9.91 19.56a5.101 5.101 0 0 1 2.24-4.22V5.42c0-.8.27-1.48.82-2.03s1.23-.84 2.03-.84c.81 0 1.49.28 2.04.83.55.56.83 1.23.83 2.03v9.92c.71.49 1.25 1.11 1.64 1.84s.58 1.53.58 2.38c0 .92-.23 1.78-.68 2.56s-1.07 1.4-1.85 1.85-1.63.68-2.56.68c-.92 0-1.77-.23-2.55-.68s-1.4-1.07-1.86-1.85-.68-1.63-.68-2.55zm1.76 0c0 .93.33 1.73.98 2.39.65.66 1.44.99 2.36.99.93 0 1.73-.33 2.4-1s1.01-1.46 1.01-2.37c0-.62-.16-1.2-.48-1.73-.32-.53-.76-.94-1.32-1.23l-.28-.14c-.1-.04-.15-.14-.15-.29V5.42c0-.32-.11-.59-.34-.81-.23-.21-.51-.32-.85-.32-.32 0-.6.11-.83.32-.23.21-.34.48-.34.81v10.74c0 .15-.05.25-.14.29l-.27.14c-.55.29-.98.7-1.29 1.23-.31.53-.46 1.1-.46 1.74zm.78 0c0 .71.24 1.32.73 1.82s1.07.75 1.76.75 1.28-.25 1.79-.75.76-1.11.76-1.81c0-.63-.22-1.19-.65-1.67-.43-.48-.96-.77-1.58-.85V9.69c0-.06-.03-.13-.1-.19a.299.299 0 0 0-.22-.1c-.09 0-.16.03-.21.08-.05.06-.08.12-.08.21v7.34c-.61.09-1.13.37-1.56.85-.43.49-.64 1.04-.64 1.68z"
        />
      </svg>
    </div>
    <div
      class="text-lg text-black"
    >
      25
      Â°
    </div>
  </div>
  <div
    class="flex flex-row space-x-1 items-center"
    data-testid="humidity"
  >
    <div
      class="h-10 w-10 undefined"
    >
      <svg
        viewBox="0 0 30 30"
        xml:space="preserve"
      >
        <path
          d="M7.56 17.19c0-.88.24-1.89.72-3.03s1.1-2.25 1.86-3.31c1.56-2.06 2.92-3.62 4.06-4.67l.75-.72c.25.26.53.5.83.72.41.42 1.04 1.11 1.88 2.09s1.57 1.85 2.17 2.65c.71 1.01 1.32 2.1 1.81 3.25s.74 2.16.74 3.03c0 1-.19 1.95-.58 2.86-.39.91-.91 1.7-1.57 2.36-.66.66-1.45 1.19-2.37 1.58-.92.39-1.89.59-2.91.59-1 0-1.95-.19-2.86-.57-.91-.38-1.7-.89-2.36-1.55-.66-.65-1.19-1.44-1.58-2.35s-.59-1.89-.59-2.93zm2.26-2.93c0 .83.17 1.49.52 1.99.35.49.88.74 1.59.74.72 0 1.25-.25 1.61-.74.35-.49.53-1.15.54-1.99-.01-.84-.19-1.5-.54-2-.35-.49-.89-.74-1.61-.74-.71 0-1.24.25-1.59.74-.35.5-.52 1.16-.52 2zm1.57 0v-.35c0-.08.01-.19.02-.33s.02-.25.05-.32.05-.16.09-.24c.04-.08.09-.15.15-.18.07-.04.14-.06.23-.06.14 0 .25.04.33.12s.14.21.17.38c.03.18.05.32.06.45s.01.3.01.52c0 .23 0 .4-.01.52s-.03.27-.06.45c-.03.17-.09.3-.17.38s-.19.12-.33.12c-.09 0-.16-.02-.23-.06a.335.335 0 0 1-.15-.18c-.04-.08-.07-.17-.09-.24-.02-.08-.04-.19-.05-.32-.01-.14-.02-.25-.02-.32v-.34zm.59 7.75h1.32l4.99-10.74h-1.35l-4.96 10.74zm4.3-2.99c.01.84.2 1.5.55 2 .35.49.89.74 1.6.74.72 0 1.25-.25 1.6-.74.35-.49.52-1.16.53-2-.01-.84-.18-1.5-.53-1.99-.35-.49-.88-.74-1.6-.74-.71 0-1.25.25-1.6.74-.36.49-.54 1.15-.55 1.99zm1.57 0c0-.23 0-.4.01-.52s.03-.27.06-.45.09-.3.17-.38.19-.12.33-.12c.09 0 .17.02.24.06.07.04.12.1.16.19.04.09.07.17.1.24s.04.18.05.32l.01.32v.69l-.01.32-.05.32-.1.24-.16.19-.24.06c-.14 0-.25-.04-.33-.12s-.14-.21-.17-.38c-.03-.18-.05-.33-.06-.45s-.01-.3-.01-.53z"
        />
      </svg>
    </div>
    <div
      class="text-lg text-black"
    >
      97
       %
    </div>
  </div>
</div>
`;
